### PR TITLE
Add ability to select program from applicant form

### PR DESCRIPTION
**Pull request by ShishckovA**

### CODE REVIEW

File: back-end/src/common/filters/universities.py [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+    )
+
+    def filter_by_campus(self, queryset, name, value):
+        # pylint: disable=unused-argument
+        return queryset.filter(faculty__campus=value)
 
     class Meta:
         model = Program
-        fields = "__all__"
+        fields = ["campus"]
```
Comment (TmLev) [inline back-end/src/common/filters/universities.py:21]: Let's filter by all fields, not just campus.

Comment (ShishckovA) [reply back-end/src/common/filters/universities.py]: OK

File: front-end/src/views/ApplicantForm/index.vue [hunk 2]
```diff
@@ -637,13 +642,35 @@ class ApplicantForm extends Vue {
...
   }
 
+  @Watch("applicantData.universityInfo.campus")
+  async onCampusChange() {
+    try {
+      const { data } = await getProgramsByCampus(
+        this.applicantData.universityInfo.campus,
+      );
+      this.fillProgramOptions(data);
+      this.applicantData.universityInfo.program = this.fields.universityInfo.program.props.options[0].value;
```
Comment (TmLev) [inline front-end/src/views/ApplicantForm/index.vue:652]: ?

File: front-end/src/views/ApplicantForm/index.vue [hunk 3]
```diff
@@ -424,6 +420,15 @@ class ApplicantForm extends Vue {
     );
   }
 
+  fillProgramOptions(data) {
+    this.fields.universityInfo.program.props.options = data.map(
```
Comment (TmLev) [inline front-end/src/views/ApplicantForm/index.vue:424]: ?

File: back-end/src/ams/models/applicants.py [hunk 4]
```diff
@@ -85,7 +85,10 @@ class Meta:
         verbose_name_plural = "Application Processes"
 
     def __str__(self) -> str:
-        return f"[{self.id}] {self.applicant.fullname} / {self.applicant.id}"
+        try:
+            return f"[{self.id}] {self.applicant.fullname} / {self.applicant.id}"
+        except Applicant.DoesNotExist:
+            return f"[{self.id}]"
```
Comment (TmLev) [inline back-end/src/ams/models/applicants.py:91]: In theory, ApplicationProcess must always have Applicant, but okay.

**Review by TmLev — APPROVED** [review]
LGTM.

**Pull request merged**